mux.js

```diff
@@ -6,40 +6,43 @@ var debug = require('debug')('multifeed')
 
 // constants
 var MULTIFEED = 'MULTIFEED'
-var PROTOCOL_VERSION = '2.0.0'
+var PROTOCOL_VERSION = '3.0.0'
+
 // extensions
 var MANIFEST = 'MANIFEST'
 var REQUEST_FEEDS = 'REQUEST_FEEDS'
+var REPLICATE_FEEDS = 'REPLICATE_FEEDS'
 
 var SupportedExtensions = [
   MANIFEST,
+  REPLICATE_FEEDS, // <-- bug in hyperprotocol? extension is never fired if declared last
   REQUEST_FEEDS
 ]
-
 // `key` - protocol encryption key
 // `opts`- hypercore-protocol opts
 function Multiplexer (key, opts) {
   if (!(this instanceof Multiplexer)) return new Multiplexer(key, opts)
-  opts = opts || {}
-  this._id = opts._id || Math.floor(Math.random() * 1000).toString(16)
-  debug(this._id + ' [REPLICATION] New mux initialized', key.toString('hex'), opts)
   var self = this
   self._opts = opts = opts || {}
-  self.extensions = opts.extensions = SupportedExtensions || opts.extensions
+  this._id = opts._id || Math.floor(Math.random() * 1000).toString(16)
```

Comment:
Will this be a problem if `opts === undefined`?

Reply:
Shouldn't be, because i tried to normalize opts to an empty hash when falsy,
 `self._opts = opts = opts || {}`
And also store a reference to the original opts as `self._opts` at the same time.

Reply:
Ah, right! Somehow missed that.